### PR TITLE
Navigation: Add app observability plugin to Observability navigation group

### PR DESCRIPTION
**Pull request by domasx2**
- Adds application observability plugin to "Observability" navigation group
- Moves "synthetic" behind "frontend"

### CODE REVIEW

File: pkg/services/navtree/navtreeimpl/applinks.go [hunk 1]
```diff
@@ -255,8 +255,9 @@ func (s *ServiceImpl) hasAccessToInclude(c *contextmodel.ReqContext, pluginID st
 func (s *ServiceImpl) readNavigationSettings() {
 	s.navigationAppConfig = map[string]NavigationAppConfig{
 		"grafana-k8s-app":                  {SectionID: navtree.NavIDMonitoring, SortWeight: 1, Text: "Kubernetes"},
-		"grafana-synthetic-monitoring-app": {SectionID: navtree.NavIDMonitoring, SortWeight: 2, Text: "Synthetics"},
+		"grafana-app-observability-app":    {SectionID: navtree.NavIDMonitoring, SortWeight: 2, Text: "Application"},
```
Comment (cyrille-leclerc) [inline pkg/services/navtree/navtreeimpl/applinks.go:258]: Do we confirm we don't annotate by "Preview" like "Application (preview)"?

**Review by connorlindsey — APPROVED** [review]

File: pkg/services/navtree/navtreeimpl/applinks.go (hunk 1 above)
Comment (domasx2) [reply pkg/services/navtree/navtreeimpl/applinks.go:258]: @cyrille-leclerc who can make the decision?

Comment (cyrille-leclerc) [reply pkg/services/navtree/navtreeimpl/applinks.go:258]: @cedricziel or @Rootzreggae ?

Comment (cedricziel) [reply pkg/services/navtree/navtreeimpl/applinks.go:258]: It resonates with me to add a "(preview)" suffix.

Comment (domasx2) [reply pkg/services/navtree/navtreeimpl/applinks.go]: added!

File: pkg/services/navtree/navtreeimpl/applinks.go [hunk 2]
```diff
@@ -255,8 +255,9 @@ func (s *ServiceImpl) hasAccessToInclude(c *contextmodel.ReqContext, pluginID st
 func (s *ServiceImpl) readNavigationSettings() {
 	s.navigationAppConfig = map[string]NavigationAppConfig{
 		"grafana-k8s-app":                  {SectionID: navtree.NavIDMonitoring, SortWeight: 1, Text: "Kubernetes"},
-		"grafana-synthetic-monitoring-app": {SectionID: navtree.NavIDMonitoring, SortWeight: 2, Text: "Synthetics"},
+		"grafana-app-observability-app":    {SectionID: navtree.NavIDMonitoring, SortWeight: 2, Text: "Application (preview)"},
 		"grafana-kowalski-app":             {SectionID: navtree.NavIDMonitoring, SortWeight: 3, Text: "Frontend"},
```
Comment (cyrille-leclerc) [inline pkg/services/navtree/navtreeimpl/applinks.go:259]: @cedricziel sorry for the late notice, should we also indicate "preview" for Frontend Observability?

Comment (Rootzreggae) [reply pkg/services/navtree/navtreeimpl/applinks.go:259]: @cyrille-leclerc My assumption has been that App-o11y is preview, but Frontend Observability isn't; they're at different stages of development, so their states aren't directly correlated; correct me if I'm wrong @cedricziel

Comment (domasx2) [reply pkg/services/navtree/navtreeimpl/applinks.go:259]: Gonna merge this, let's figure out Frontend Observability separately :)

Comment (cyrille-leclerc) [reply pkg/services/navtree/navtreeimpl/applinks.go:259]: Thanks, there is uncertainty on the GA date for Frontend O11y due to the pricing question. @cedricziel what would be your recommendation?

**Pull request merged**